bazel_tools/scala.bzl

```diff
@@ -201,6 +201,121 @@ def _create_scala_source_jar(**kwargs):
             srcs = kwargs["srcs"],
         )
 
+def _scaladoc_jar_impl(ctx):
+    srcFiles = [
+        src.path
+        for src in ctx.files.srcs
+    ]
+    pluginPaths = []
+    for p in ctx.attr.plugins:
+        if hasattr(p, "path"):
+            pluginPaths.append(p)
+        elif hasattr(p, "scala"):
+            pluginPaths.extend([j.class_jar for j in p.scala.outputs.jars])
+        elif hasattr(p, "java"):
```

Comment:
In `transitive_deps` below you had to switch from `.java` to `[JavaInfo]`, right? Do plugins still have `.java`, or should this part also be switched to use `[JavaInfo]`?

Reply:
The plugins here will be pure java plugins. This could was lifted from 'rules_scala' so I'm inclined to leave it.

Reply:
Makes sense. Thanks for clarifying.